### PR TITLE
Changed great-circle distance function

### PR DESCRIPTION
**Pull request by mjziebarth**
to a more numerically stable one (based on Vincenty formula). 
Added references while doing so.

Edit: This is also the same method that geopy's great-circle distance is based off of:
[distance.py](https://github.com/geopy/geopy/blob/master/geopy/distance.py)

### CODE REVIEW

**Comment by rth** [comment]
@mjziebarth Thanks for the PR. 

Could you add another unit tests to check that we obtain the same results as in geopy? Say for some input, compute the result with geopy externally, and then write a test checking that PyKrige produces the same (hard-coded) value, then indicate in comments how that value was computed with geopy? What do you think?

**Comment by mjziebarth** [comment]
@rth Yes, created some reference values and added them (including a comment how to recreate them) to the test routine.

Currently, that's 200 pseudorandom positions on a sphere created from a constant seed. Then, a distance matrix is created and some properties of that matrix are compared with the values obtained from a geopy-generated distance matrix.

Thoughts?

File: pykrige/test.py [hunk 1]
```diff
@@ -1388,9 +1388,59 @@ def test_uk3d_functional_drift(self):
...
+                                   [179.39827328486047, # max
+                                    0.49781321229536524, # min != 0
+                                    3577568.5670485105, # sum
+                                    39.42686179130957, # std
+                                    90.011132985696634, # median
+                                    5.2851101415176212, # percentile
+                                    106.42435552834375,
+                                    75.241117692324522,
+                                    109.48819700842446,
+                                    40.639971012523361])
```
Comment (rth) [inline pykrige/test.py]: How about computing geopy for just 10 random points (i.e. `N=10`) and then comparing that pykridge produces the same results for these points with `assert_allclose`. 10 points should be enough to check that it's working correctly, and it's conceptually easier to understand than comparing the min, max, percentile etc of d. What do you think?

Comment (mjziebarth) [inline pykrige/test.py]: I first thought about that, but here's why I didn't do it: In that case, even for small N, there would be lots of data to include in test.py. Intuitively, I found that a lot of text added without much a benefit. But I'm not too sure about it, so I could add a plain reference if you wish.

Comment (rth) [inline pykrige/test.py]: Aww, right is it a `NxN` matrix.. Well we can just use `N=2` then? I mean this is just to check that the output is not off by orders of magnitude, so even checking on a single value could be enough...

Comment (mjziebarth) [inline pykrige/test.py]: Okay, I can do that. Would also just use predefined coordinate pairs then and remove the numpy random code.

**Comment by rth** [comment]
Everything looks good to me. Thanks for the PR!

**Pull request merged**